session-replay/Sources/DatadogSessionReplay/Recorder/TouchSnapshotProducer/TouchSnapshot/TouchSnapshot.swift

```diff
@@ -14,12 +14,20 @@ internal struct TouchSnapshot {
         /// An unique identifier of the touch. It persists throughout a multi-touch sequence (it is created on "touch down",
         /// continues thru "touch move" and ends in "touch up").
         let id: TouchIdentifier
+        /// Phase of the touch as distinguished in session replay.
+        let phase: TouchPhase
         /// A time of recording this touch
         let date: Date
         /// The position of this touch in application window.
         let position: CGPoint
     }
 
+    enum TouchPhase {
+        case down
+        case move
+        case up
+    }
```

Comment:
Why not using the `PointerEventType` instead?

Reply:
I prefer to have a clear separation between layers, hence:
* `TouchSnapshot ` and `TouchSnapshot .TouchPhase` - are part of the Recorder,
* `PointerEventType` - is part of the Writer (especially data models we generate).

Using `PointerEventType` in `Recorder` would work and prevent model code duplication, but might be IMO a wrong abstraction. "Touch phase" that we record could be different from what we send at the end (e.g. iOS differentiates more than "up" phase: `.ended`, `.cancelled`, `.regionExited` and if for any reason we need to handle it differently, separate model will be more convenient).